Duckling/Time/EN/Rules.hs

```diff
@@ -1613,11 +1613,11 @@ ruleEndOrBeginningOfMonth :: Rule
 ruleEndOrBeginningOfMonth = Rule
   { name = "at the beginning|end of <named-month>"
   , pattern =
-    [ regex "(at the )?(beginning|start|end) of"
+    [ regex "(at )?(the )?(beginning|start|end) of"
     , Predicate isAMonth
     ]
   , prod = \tokens -> case tokens of
-      (Token RegexMatch (GroupMatch (_:match:_)):Token Time td:_) -> do
+      (Token RegexMatch (GroupMatch (_:_:match:_)):Token Time td:_) -> do
```

Comment:
If it doesn't introduce any issues, could you make the same change to `ruleEndOrBeginningOfYear` below? It has the same issue where "the" isn't part of the extraction. This would enable something like "between the end of 2021 and the start of 2022."